server/src/main/java/org/elasticsearch/index/translog/TranslogReader.java

```diff
@@ -70,6 +75,30 @@ public static TranslogReader open(
         return new TranslogReader(checkpoint, channel, path, header);
     }
 
+    /**
+     * Closes current reader and creates new one with new checkoint and same file channel
+     */
+    TranslogReader closeIntoTrimmedReader(long belowTerm, long aboveSeqNo, ChannelFactory channelFactory) throws IOException {
+        if (!closed.get()
+            && getPrimaryTerm() < belowTerm
+            && checkpoint.maxSeqNo != SequenceNumbers.NO_OPS_PERFORMED
```

Comment:
I don't think we need this optimization, it's rare we do this and it just adds complexity. Can you remove please?